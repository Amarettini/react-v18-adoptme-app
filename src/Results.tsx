import { FunctionComponent } from "react";
import Pet from "./Pet";
import { Pet as PetType } from "./APIResponsesTypes";

interface Props {
  pets: PetType[];
}

const Results: FunctionComponent<Props> = ({ pets }) => {
  return (
    <div className="search">
      {!pets.length ? (
        <h1>No Pets Found</h1>
      ) : (
        pets.map((pet) => (
          <Pet
            animal={pet.animal}
            key={pet.id}
            name={pet.name}
            breed={pet.breed}
            images={pet.images}
            location={`${pet.city}, ${pet.state}`}
            id={pet.id}
          />
        ))
      )}
    </div>
  );
};

export default Results;
