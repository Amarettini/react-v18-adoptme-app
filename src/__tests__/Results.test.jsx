import { expect, test } from "vitest";
import { render } from "@testing-library/react";
import Results from "../components/Results";

test("renders correctly with no pets", () => {
  const { asFragment } = render(<Results pets={[]} />);
  expect(asFragment()).toMatchSnapshot();
});
