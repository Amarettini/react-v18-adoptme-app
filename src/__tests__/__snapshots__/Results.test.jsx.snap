// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly with no pets 1`] = `
<DocumentFragment>
  <div
    class="search"
  >
    <h1>
      No Pets Found
    </h1>
  </div>
</DocumentFragment>
`;
